### ISSUE TITLE
Questions about the paper

### ISSUE THREAD

**Issue by wamiq-reyaz** [he-dhamo/graphto3d#2]
The title of the paper says end-to-end. 

Is it true that you train the DeepSDF model together with the GCNs?

If so where can I find the architecture of the shape Encoder <img src="https://render.githubusercontent.com/render/math?math=E_{gen}">
? What is the input modality for the shape - is it a point cloud, SDF, voxel grid? Is that encoder pretrained? Otherwise training with <img src="https://render.githubusercontent.com/render/math?math=||e_i^s - \hat{e}_i^s||_{1}"> loss makes no sense. 

I can't figure out where in the paper this is discussed.

Another thing that is unclear to me is the labelling of the latent graph - is the <img src="https://render.githubusercontent.com/render/math?math=\mu, \sigma"> generated per class and per node? I am guessing only the *object* nodes get any latent while the edge/relationship nodes are processed by the GCN to learn their meaning and do not have any KL term. But instead just propagate as are? Eq 1,2,3 are a bit unclear to me.

Thanks!

**Comment by he-dhamo** [comment]
As we write in the paper (e.g. Section 4.2), our GCN-based VAE model doesn't build on shapes <img src="https://render.githubusercontent.com/render/math?math=s_i">  directly (SDF, point clouds, etc) but on shape latent encodings <img src="https://render.githubusercontent.com/render/math?math=e_i^s"> , which are 1D vectors, and therefore the L_1 loss (applied on shape encodings) makes sense. 

The term end-to-end here refers to fully-leaned (as opposed to retrieved for example), but in practice we train the AtlasNet or DeepSDF model beforehand and pass the respective shape embeddings to our VAE model. So far we provide the learned weights for AtlasNet. 

There is no KL term on relationships. The semantic graph (class nodes and edges) is a condition that controls the generation of object shapes and boxes, therefore the learned latent basically needs to describe object nodes.
